### PR TITLE
Feature/actions

### PR DESCRIPTION
**Pull request by Chesteer89**

### CODE REVIEW

**Comment by Chesteer89** [comment]
Preview: image

File: chat-components/src/main/java/com/pubnub/components/chat/service/message/action/DefaultMessageReactionServiceImpl.kt [hunk 1]
```diff
@@ -21,14 +22,15 @@ import timber.log.Timber
     FlowPreview::class,
     DelicateCoroutinesApi::class,
 )
-class DefaultMessageReactionService(
+class DefaultMessageReactionServiceImpl(
```
Comment (bartk) [inline chat-components/src/main/java/com/pubnub/components/chat/service/message/action/DefaultMessageReactionServiceImpl.kt:25]: Is that `Impl` suffix needed?

File: chat-components/ui/src/main/java/com/pubnub/components/chat/ui/component/common/IconThemeDefaults.kt [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+import androidx.compose.ui.graphics.Shape
+import androidx.compose.ui.graphics.vector.ImageVector
+import androidx.compose.ui.unit.dp
+
+object IconThemeDefaults {
+    @Composable
+    fun icon(
+        icon: ImageVector? = null,
+        shape: Shape = CircleShape,
+        tint: Color = MaterialTheme.colors.onBackground.copy(alpha = ContentAlpha.medium),//LocalContentColor.current.copy(alpha = ContentAlpha.medium),
```
Comment (bartk) [inline chat-components/ui/src/main/java/com/pubnub/components/chat/ui/component/common/IconThemeDefaults.kt:20]: commented code

**Review by bartk — APPROVED** [review]
:+1 although I placed some comments and questions.

File: chat-components/framework/build.gradle [hunk 3]
```diff
@@ -13,7 +13,7 @@ android {
         minSdk = 21
         targetSdk = 31
         versionCode = 1
-        versionName = "0.2.0"
+        versionName = "0.3.0-rc28"
```
[the remark this answers is not in the document]
Comment (Chesteer89) [reply chat-components/framework/build.gradle:16]: Decided to keep it splitted, it should be bumped only when the changes affected the project

File: chat-components/src/main/java/com/pubnub/components/chat/service/message/action/DefaultMessageReactionServiceImpl.kt (hunk 1 above)
Comment (Chesteer89) [reply chat-components/src/main/java/com/pubnub/components/chat/service/message/action/DefaultMessageReactionServiceImpl.kt:25]: Not really, just a convention. We can remove it everywhere

File: chat-components/database/src/main/java/com/pubnub/components/repository/channel/ChannelRepository.kt [hunk 4]
```diff
@@ -16,7 +16,7 @@ interface ChannelRepository<DB : Channel, Data : Channel> {
     ): PagingSource<Int, Data>
 
     suspend fun getList(): List<Data>
-    suspend fun add(vararg channel: DB)
+    suspend fun insertOrUpdate(vararg channel: DB)
```
[the remark this answers is not in the document]
Comment (Chesteer89) [reply chat-components/database/src/main/java/com/pubnub/components/repository/channel/ChannelRepository.kt]: Resolved: IN / OUT parameters

**Comment by Chesteer89** [comment]
Custom emoji:
custom_emoji mp4

**Pull request merged**